lib/logger/src/main/scala/org.bitbucket.pshirshov.izumi/logger/LogSink.scala

```diff
@@ -0,0 +1,38 @@
+package org.bitbucket.pshirshov.izumi.logger
+
+import scala.annotation.tailrec
+
+trait LogSink {
+
+  def messageParser(msg: Log.Message): Map[String, Any] = {
+    def truncateKey(s: String) = {
+      "(\\w+)=".r.findFirstIn(s).map(_.dropRight(1))
+    }
+    @tailrec
+    def parse(keysIterator: Iterator[String]
```

Comment:
You may avoid recursion here. It may be a premature optimization for now though it's easy to parse everything during a single pass.